### PR TITLE
REMOVE most of the NGSIv1 deprecated requests

### PR DESCRIPTION
**Pull request by fgalan**
Pending:

- [x] Disable involved .tests
    - `for f in $(cat test-using-only-v1.txt) ; do git mv $f $f.DISABLED ; done` - commit [ab6a34c](https://github.com/telefonicaid/fiware-orion/pull/4345/commits/ab6a34cbff1c03ef4738ac6b462714a958630bdb)
    - `for f in $(cat test-using-both-v1-and-v2.txt) ; do git mv $f $f.DISABLED ; done` - commit [6e10420](https://github.com/telefonicaid/fiware-orion/pull/4345/commits/6e1042074cbc0bc49601b7cf8775fc79273ed804)
    - More fixes done in other commits...
- [x] Check `GET /statistics?options=fullCounters` (so only remaining NGSIv1/NGSI10 route appear) - commit [8ce8cf2](https://github.com/telefonicaid/fiware-orion/pull/4345/commits/8ce8cf2d6b3bfec15194ccf0da71f71c464d4855)
- [x] Recover "over-disabled" tests (i.e. tests now .DISABLED which only NGSIv1 operations used are some of the still supported ones)
- [x] Create tech debt issue regarding the analysis of the disabled tests in this PR - issue https://github.com/telefonicaid/fiware-orion/issues/4349
- [x] Review versions numbers and dates in deprecated.md file - commit https://github.com/telefonicaid/fiware-orion/pull/4345/commits/e726415bafa0e09950f46ed718f7eb1bccb160cd

### CODE REVIEW

**Comment by fgalan** [comment]
CNR in this PR include the 7 legacy operations that will remain after this cleanup.

In the case if may be useful, this is the explicit list of the 71 operations (verb + url) that have been removed:

```
GET /v1/registry/contextEntities/{id}
POST /v1/registry/contextEntities/{id}
GET /v1/registry/contextEntities/{id}/attributes
POST /v1/registry/contextEntities/{id}/attributes
GET /v1/registry/contextEntities/{id}/attributes/{name}
POST /v1/registry/contextEntities/{id}/attributes/{name}
GET /v1/registry/contextEntityTypes/{type}
POST /v1/registry/contextEntityTypes/{type}
GET /v1/registry/contextEntityTypes/{type}/attributes
POST /v1/registry/contextEntityTypes/{type}/attributes
GET /v1/registry/contextEntityTypes/{type}/attributes/{name}
POST /v1/registry/contextEntityTypes/{type}/attributes/{name}
GET /v1/contextEntities/{id}
POST /v1/contextEntities/{id}
GET /ngsi10/contextEntities/{id}
POST /ngsi10/contextEntities/{id}
PUT /ngsi10/contextEntities/{id}
DELETE /ngsi10/contextEntities/{id}
GET /v1/contextEntities/{id}/attributes/
POST /v1/contextEntities/{id}/attributes/
PUT /v1/contextEntities/{id}/attributes/
DELETE /v1/contextEntities/{id}/attributes/
GET /ngsi10/contextEntities/{id}/attributes/
POST /ngsi10/contextEntities/{id}/attributes/
PUT /ngsi10/contextEntities/{id}/attributes/
DELETE /ngsi10/contextEntities/{id}/attributes/
POST /v1/contextEntities/{id}/attributes/{name}
PUT /v1/contextEntities/{id}/attributes/{name}
DELETE /v1/contextEntities/{id}/attributes/{name}
GET /ngsi10/contextEntities/{id}/attributes/{name}
POST /ngsi10/contextEntities/{id}/attributes/{name}
PUT /ngsi10/contextEntities/{id}/attributes/{name}
DELETE /ngsi10/contextEntities/{id}/attributes/{name}
GET /v1/contextEntityTypes/{type}
GET /ngsi10/contextEntityTypes/{type}
GET /v1/contextEntityTypes/{type}/attributes/
GET /ngsi10/contextEntityTypes/{type}/attributes/
GET /v1/contextEntityTypes/{type}/attributes/{name}
GET /ngsi10/contextEntityTypes/{type}/attributes/{name}
PUT /v1/contextSubscriptions/{id}
DELETE /v1/contextSubscriptions/{id}
PUT /ngsi10/contextSubscriptions/{id}
DELETE /ngsi10/contextSubscriptions/{id}
GET /v1/contextTypes
GET /v1/contextType/{type}
GET /v1/contextEntitites
POST /v1/contextEntitites
GET /v1/contextEntities/type/{type}/id/{id}
POST /v1/contextEntities/type/{type}/id/{id}
PUT /v1/contextEntities/type/{type}/id/{id}
DELETE /v1/contextEntities/type/{type}/id/{id}
GET /v1/contextEntities/type/{type}/id/{id}/attributes/{name}
POST /v1/contextEntities/type/{type}/id/{id}/attributes/{name}
PUT /v1/contextEntities/type/{type}/id/{id}/attributes/{name}
DELETE /v1/contextEntities/type/{type}/id/{id}/attributes/{name}
GET /v1/registry/contextEntities/type/{type}/id/{id}
POST /v1/registry/contextEntities/type/{type}/id/{id}
GET /v1/registry/contextEntities/type/{type}/id/{id}/attributes/{name}
POST /v1/registry/contextEntities/type/{type}/id/{id}/attributes/{name}
POST /v1/registry/registerContext
POST /v1/registry/discoverContextAvailability
POST /v1/subscribeContext
POST /v1/contextSubscriptions
POST /ngsi10/subscribeContext
POST /ngsi10/contextSubscriptions
POST /v1/updateContextSubscription
POST /ngsi10/updateContextSubscription
POST /v1/unsubscribeContext
POST /ngsi10/unsubscribeContext
POST /v1/notifyContext
POST /ngsi10/notifyContext
```

**Comment by fgalan** [comment]
Useful commant to (coarsely) analyze NGSI usages:

```
grep -R -i -E -l 'orionCurl.*v1|orionCurl.*ngsi10' cases/ | sort > test-using-ngsiv1.txt
grep -R -i -E -l 'orionCurl.*v2' cases/ | sort > test-using-ngsiv2.txt
comm -12 test-using-ngsiv1.txt test-using-ngsiv2.txt > test-using-both-v1-and-v2.txt
comm -23 test-using-ngsiv1.txt test-using-ngsiv2.txt > test-using-only-v1.txt

**Comment by fgalan** [comment]
```
$ wc -l test-using-*.txt
   157 test-using-both-v1-and-v2.txt
   429 test-using-ngsiv1.txt
  1102 test-using-ngsiv2.txt
   272 test-using-only-v1.txt
  1960 total
```

**Comment by fgalan** [comment]
Manually remove cases/0000_statistics_operation/statistics_with_counters.test from test-using-both-v1-and-v2.txt (so count is now 156)

File: doc/manuals/deprecated.md [hunk 1]
```diff
@@ -27,6 +27,14 @@ A list of deprecated features and the version in which they were deprecated foll
   Use NGSIv2 API instead.
     * Context availability subscriptions (AKA NGSI9 subscriptions), as part of NGSIv1, in Orion 2.0.0
       (removed in Orion 2.6.0)
+    * All the other NGSIv1 operations except the following ones:
```
[the remark this answers is not in the document]
Comment (fgalan) [reply doc/manuals/deprecated.md]: Fixed in [761d2cb](https://github.com/telefonicaid/fiware-orion/pull/4345/commits/761d2cb4617ae85536f791eaf7bf4f3d26c8e1d5)

**Comment by fgalan** [comment]
> The result (sorted) is this:

False positives due to /v1/updateContext also matchs /v1/updateContextSubscription:

```
./cases/0000_large_requests/large_update_context_subscription.test
./cases/0000_ngsi10_subscriptions/update_context_subscription.test
```

False positive due to /v1/notifyContext in "url" (for subscription):

```
./cases/0001_federation/federation_location_ngsiv1.test
```

File: test/functionalTest/cases/0000_deprecated_checkings/log_deprecate_warning.test [hunk 2]
```diff
@@ -50,9 +50,9 @@ echo
 echo
 
 
-echo "02. GET /v1/contextEntities/E"
-echo "============================="
-orionCurl --url /v1/contextEntities/E
+echo "02. GET /v1/contextEntities/E/attributes/A"
+echo "=========================================="
+orionCurl --url /v1/contextEntities/E/attributes/A
```
Comment (fgalan) [inline test/functionalTest/cases/0000_deprecated_checkings/log_deprecate_warning.test:55]: The NGSIv1 operation without payload we used in this test was deprecated, so we have changed to another one.

NTC (informative)

**Comment by fgalan** [comment]
After internal discussion with @mapedraza we have decided to postpone this PR after releasing Orion 3.9.0 (so in Orion 3.9.0 the new `-logDeprecated` is available for the full set of NGSIv1 deprecated operations before doing the removal).

Versions and dates would need to be modified in deprecated.md before merging this PR.

**Comment by fgalan** [comment]
@fisuda this PR does some modifications to deprecated.md file. It would be great if you could sync also the Japanese translation, please. Thanks!

**Comment by fisuda** [comment]
I sent the PR #4353. Thanks.

**Pull request merged**